pkgs/by-name/xc/xcursor-pro/package.nix

```
@@ -0,0 +1,48 @@
+{
+  stdenvNoCC,
+  lib,
+  fetchurl,
+  variants ? [
+    "Dark"
+    "Light"
+    "Red"
+  ],
+}:
+stdenvNoCC.mkDerivation rec {
+  pname = "xcursor-pro";
+  version = "2.0.2";
+  srcs = map (
+    variant:
+    fetchurl {
+      url = "https://github.com/ful1e5/XCursor-pro/releases/download/v${version}/XCursor-Pro-${variant}.tar.xz";
+      hash =
+        {
+          Dark = "sha256-lqA7rt1+yqLRv99XccAcYhEMiFmmlgmmL1Sr57lYJ4s=";
+          Light = "sha256-irYZLu8E1VttcUXQ+7XfUKxKyT1MbesG/b03qRBubac=";
+          Red = "sha256-NWkWHTMbC6/AD5P19d65F3sTqxD+5ZLmtsN/5ljFMAM=";
+        }
+        .${variant};
+    }
+  ) variants;
+
+  sourceRoot = ".";
+
+  installPhase = ''
+    for theme in XCursor-Pro-{${lib.concatStringsSep "," variants}}; do
+      mkdir -p $out/share/icons/$theme/cursors
+      cp -a $theme/cursors/* $out/share/icons/$theme/cursors/
+      install -m644 $theme/index.theme $out/share/icons/$theme/index.theme
+    done
+  '';
```

```suggestion
  installPhase = ''
    runHook preInstall
    for theme in XCursor-Pro-{${lib.concatStringsSep "," variants}}; do
      mkdir -p $out/share/icons/$theme/cursors
      cp -a $theme/cursors/* $out/share/icons/$theme/cursors/
      install -m644 $theme/index.theme $out/share/icons/$theme/index.theme
    done
    runHook postInstall
  '';
```

The `installPhase` should call the pre- and post-install hooks.